src/context/tasks.context.tsx

```diff
@@ -0,0 +1,57 @@
+import fetch from "@/helperFunctions/fetch";
+import React, { createContext, useContext ,useState, useEffect} from "react";
+import levelType from "@/interfaces/level.type";
+import tagType from "@/interfaces/tag.type";
+
+export type TaskContextType = {
+    taskTags: tagType[] | null,
+    taskLevels: levelType[] | null
+}
+
+const TasksContext = createContext<TaskContextType>({
+    taskTags: [],
+    taskLevels: []
+})
+
+export const TasksProvider = ({children }: { children: React.ReactNode })=> {
+    const [taskTags, setTaskTags] = useState<tagType[] | null>(null)
+    const [taskLevels, setTaskLevels] = useState<levelType[] | null>(null)
+    const ALL_TAGS_URL = `${process.env.NEXT_PUBLIC_BASE_URL}/tags`;
+    const ALL_LEVELS_URL = `${process.env.NEXT_PUBLIC_BASE_URL}/levels`;
```

Comment:
Could we look into having one file for all the links,
Maybe you can go ahead and create ticket to address this 
cc: @Maheima

Reply:
moved to `constants -> urls.js`